### PR TITLE
Feature/pep8

### PR DESCRIPTION
**Pull request by eileen-kuehn**
This pull request includes configuration and integration of flake8. I further added the required changes to make the tests succeed.

### CODE REVIEW

File: .pep8speaks.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+scanner:
+  diff_only: False  # If False, the entire file touched by the Pull Request is scanned for errors. If True, only the diff is scanned.
+  linter: flake8
+
+flake8:
+  max-line-length: 88  # flake8-bugbear uses 80*1.1
+  ignore:
+    - W503
```
Comment (maxfischer2781) [inline .pep8speaks.yml]: Add new line at end of file.
```suggestion
    - W503

```

File: lapis/drone.py [hunk 2]
```diff
@@ -11,20 +11,24 @@ class ResourcesExceeded(Exception):
...
+    def __init__(self, scheduler, pool_resources: dict,
+                 scheduling_duration: float, exclusive: bool = False,
+                 ignore_resources: list = None):
         """
         :param scheduler:
         :param pool_resources:
         :param scheduling_duration:
-        :param exclusive: Determines if the drone is used exclusively by jobs in sequential order
+        :param exclusive: Determines if the drone is used exclusively by jobs
+        in sequential order
```
Comment (maxfischer2781) [inline lapis/drone.py]: I recommend aligning the body of Sphinx fields internally, not with the key.

```suggestion
                          in sequential order
```

File: lapis/drone.py [hunk 3]
```diff
@@ -11,20 +11,24 @@ class ResourcesExceeded(Exception):
...
         """
         super(Drone, self).__init__()
         self.scheduler = scheduler
         self.pool_resources = pool_resources
         self.resources = {resource: 0 for resource in self.pool_resources}
         if ignore_resources:
-            self._valid_resource_keys = [resource for resource in self.pool_resources if resource not in ignore_resources]
+            self._valid_resource_keys = [
+                resource for resource in self.pool_resources
+                if resource not in ignore_resources]
```
Comment (maxfischer2781) [inline lapis/drone.py]: The next (dict) comprehension has a closing mark on the next line, this (list) comprehension has a closing mark trailing. I recommend using one style for comprehensions; I prefer using an extra line for the closing mark.

File: lapis/drone.py [hunk 4]
```diff
@@ -104,23 +113,27 @@ def _add_resources(self, keys: list, target: dict, source: dict, alternative_sou
...
         """
         Method manages to start a job in the context of the given drone.
-        The job is started independent of available resources. If resources of drone are exceeded, the job is killed.
+        The job is started independent of available resources. If resources of
+        drone are exceeded, the job is killed.
 
         :param job: the job to start
-        :param kill: if True, a job is killed when used resources exceed requested resources
+        :param kill: if True, a job is killed when used resources exceed
+        requested resources
```
Comment (maxfischer2781) [inline lapis/drone.py]: See above note on aligning in sphinx docstrings.

```suggestion
                     requested resources
```

File: lapis/job.py [hunk 5]
```diff
@@ -18,40 +15,41 @@ def job_demand(simulator):
...
+        Definition of a job that uses a specified amount of resources `used_resources`
+        over a given amount of time, `walltime`. A job is described by its user
+        via the parameter `resources`. This is a user prediction and is expected
+        to deviate from `used_resources`.
 
         :param resources: Requested resources of the job
         :param used_resources: Resource usage of the job
-        :param in_queue_since: Time when job was inserted into the queue of the simulation scheduler
+        :param in_queue_since: Time when job was inserted into the queue of the
+        simulation scheduler
```
Comment (maxfischer2781) [inline lapis/job.py]:
```suggestion
                               simulation scheduler
```

File: lapis/job_io/htcondor.py [hunk 6]
```diff
@@ -4,46 +4,51 @@
...
         if float(row[used_resource_name_mapping["walltime"]]) <= 0:
-            logging.getLogger("implementation").warning("removed job from htcondor import", row)
+            logging.getLogger("implementation").warning(
+                "removed job from htcondor import", row)
             continue
         resources = {}
         for key, original_key in resource_name_mapping.items():
             try:
-                resources[key] = float(row[original_key]) * unit_conversion_mapping.get(original_key, 1)
+                resources[key] = float(row[original_key]) \
```
Comment (maxfischer2781) [inline lapis/job_io/htcondor.py]: Is the line break needed here? The enclosing parentheses might make it redundant.

File: lapis/pool.py [hunk 7]
```diff
@@ -126,15 +129,19 @@ def __repr__(self):
...
-    initialises the `capacity` of given drones with initialised `resources`.
-
-    :param capacity: Maximum number of pools that can be instantiated within the pool
-    :param resources: Dictionary of resources available for each pool instantiated within the pool
+    A static pool does not react on changing conditions regarding demand,
+    allocation and utilisation but instead initialises the `capacity` of given
+    drones with initialised `resources`.
+
+    :param capacity: Maximum number of pools that can be instantiated within
+    the pool
```
Comment (maxfischer2781) [inline lapis/pool.py]:
```suggestion
                     the pool
```

File: lapis/pool.py [hunk 8]
```diff
@@ -126,15 +129,19 @@ def __repr__(self):
...
+    :param resources: Dictionary of resources available for each pool
+    instantiated within the pool
```
Comment (maxfischer2781) [inline lapis/pool.py]:
```suggestion
                      instantiated within the pool
```

File: lapis/pool_io/htcondor.py [hunk 9]
```diff
@@ -5,34 +5,37 @@
...
+}, pool_type: Callable = Pool, make_drone: Callable = None):
     """
-    Load a pool configuration that was exported via htcondor from files or iterables
+    Load a pool configuration that was exported via htcondor from files or
+    iterables
 
     :param iterable: an iterable yielding lines of CSV, such as an open file
-    :param resource_name_mapping: Mapping from given header names to well-defined resources in simulation
+    :param resource_name_mapping: Mapping from given header names to well-defined
+    resources in simulation
```
Comment (maxfischer2781) [inline lapis/pool_io/htcondor.py]:
```suggestion
                                  resources in simulation
```

File: lapis/pool_io/machines.py [hunk 10]
```diff
@@ -5,23 +5,28 @@
...
     """
-    Load a pool configuration that was exported via htcondor from files or iterables
+    Load a pool configuration that was exported via htcondor from files or
+    iterables
 
+    :param make_drone: The callable to create the drone
     :param iterable: an iterable yielding lines of CSV, such as an open file
-    :param resource_name_mapping: Mapping from given header names to well-defined resources in simulation
+    :param resource_name_mapping: Mapping from given header names to well-defined
+    resources in simulation
```
Comment (maxfischer2781) [inline lapis/pool_io/machines.py]:
```suggestion
                                  resources in simulation
```

File: setup.cfg [hunk 11]
```diff
@@ -1,2 +1,9 @@
 [aliases]
 test=pytest
+
+[flake8]
+statistics = True
+max-line-length = 80
+ignore = E501, B008, B011, W503
+select = C,E,F,W,B,B9
+exclude = docs,.svn,CVS,.bzr,.hg,.git,__pycache__,.tox,.eggs,*.egg
```
Comment (maxfischer2781) [inline setup.cfg]: Newline at end of file.
```suggestion
exclude = docs,.svn,CVS,.bzr,.hg,.git,__pycache__,.tox,.eggs,*.egg

```

File: lapis/drone.py (hunk 3 above)
Comment (eileen-kuehn) [reply lapis/drone.py]: I added another commit to fix this and also put the closing mark on a new line.

**Pull request merged**